components/ModEditHealthcareProfessionalSection.vue

```diff
@@ -238,52 +204,114 @@ const specialtyList = Object.values(Specialty) as Specialty[]
+const isHealthcareProfessionalInitialized: Ref<boolean> = ref(false)
+
+// These are component refs to enable the multi select to work properly
+const healthcareProfessionalAcceptedInsurancesArray: Ref<Array<Insurance>> = ref([])
+const healthcareProfessionalDegreesArray: Ref<Array<Degree>> = ref([])
+const healthcareProfessionalSpecialtiesArray: Ref<Array<Specialty>> = ref([])
+const healthcareProfessionalSpokenLanguages: Ref<Array<Locale>> = ref([])
 
 const handleLocalizedName = () => {
     const localizedNameToAdd: LocalizedNameInput = {
-        firstName: healthcareProfessionalSectionFields.localizedFirstName,
-        lastName: healthcareProfessionalSectionFields.localizedLastName,
-        locale: healthcareProfessionalSectionFields.nameLocale || Locale.EnUs,
-        middleName: healthcareProfessionalSectionFields.localizedMiddleName
+        firstName: sectionFields.localizedFirstName,
+        lastName: sectionFields.localizedLastName,
+        locale: sectionFields.nameLocale || Locale.EnUs,
+        middleName: sectionFields.localizedMiddleName
     }
 
     if (localizedNameToAdd.firstName
-      && localizedNameToAdd.lastName) {
-        professionalNameArray.push(localizedNameToAdd)
-        healthcareProfessionalSectionFields.localizedFirstName = ''
-        healthcareProfessionalSectionFields.localizedLastName = ''
-        healthcareProfessionalSectionFields.localizedMiddleName = ''
-        healthcareProfessionalSectionFields.nameLocale = Locale.EnUs
+      && localizedNameToAdd.lastName
+      && localizedNameToAdd.firstName.length > 1
+      && localizedNameToAdd.lastName.length > 1) {
+        sectionFields.healthcareProfessionalNameArray.push(localizedNameToAdd)
+        sectionFields.localizedFirstName = ''
+        sectionFields.localizedLastName = ''
+        sectionFields.localizedMiddleName = ''
+        sectionFields.nameLocale = Locale.EnUs
     }
 }
 
-const handleRemoveHealthcareProfessionalName = (index: number) => {
-    professionalNameArray.splice(index, 1)
-}
+onBeforeMount(async () => {
+    isHealthcareProfessionalInitialized.value = false
+
+    /**
+    Set the variable to useToast when the before the component mounts
+    since vue-taostification is only available on the client.
+    If not done this way the build fails
+     */
+    toast = useToast()
+
+    // Wait for the route to be fully resolved
+    await nextTick()
+    // Ensure the route param `id` is available before proceeding
+    const id = route.params.id
+
+    if (!id) {
+        console.error(t('modHealthcareProfessionalSection.errorMessageHealthcareProfessionalId'))
+        toast.error(t('modHealthcareProfessionalSection.errorMessageHealthcareProfessionalId'))
+        return
+    }
+
+    if (!healthcareProfessionalsStore.healthcareProfessionalsData.length) {
+        await healthcareProfessionalsStore.getHealthcareProfessionals()
+    }
+
+    healthcareProfessionalsStore.selectedHealthcareProfessionalId = id as string
```

Comment:
looking at this, I just thought of a way to simplify this. 
This id should be set within the `setSelectedHealthcareProfessional` function so the caller doesn't have to do it.